content/blog/getting-started-with-spark-on-mapr-sandbox.md

```diff
@@ -0,0 +1,217 @@
+
+```Java
+        String inputFile = args[0];
+        JavaRDD <string>input = sc.textFile(inputFile);</string>
+```
+
+**Transform input RDD with flatMap**
+
+To split the input text into separate words, we use the [flatMap(func)](https://spark.apache.org/docs/1.3.0/programming-guide.html) RDD transformation, which returns a new RDD formed by passing each element of the source through a function. The String split function is applied to each line of text, returning an RDD of the words in the input RDD:
+
+![](/img/tutorial_spark5.png)
+
+```Java
+         // map/split each line to multiple words
+        **JavaRDD<String> words = input.flatMap(**
+                new FlatMapFunction<String, String>() {
+                    public Iterable<String> call(String x) {
+                        return Arrays.asList(x.split(" "));
+                    }
+                }
+        );
+```
+
+**Transform words RDD with map**
+
+We use the [map(func)](https://spark.apache.org/docs/1.3.0/programming-guide.html) to transform the words RDD into an RDD of (word, 1) key-value pairs:
+
+![](/img/tutorial_spark6.png)
+
+```Java
+        JavaPairRDD<String, Integer> wordOnePairs = words.mapToPair(
+                new PairFunction<String, String, Integer>() {
+                    public Tuple2<String, Integer> call(String x) {
+                        return new Tuple2(x, 1);
+                    }
+                }
+        );
+```
+
+**Transform wordOnePairs RDD with reduceByKey**
+
+To count the number of times each word occurs, we combine the values (1) in the wordOnePairs with the same key (word) using [reduceByKey(func)](https://spark.apache.org/docs/1.3.0/programming-guide.html), This transformation will return an RDD of (word, count) pairs where the values for each word are aggregated using the given reduce function func x+y:
+
+![](/img/tutorial_spark7.png)
+
+```Java
+        // reduce add the pairs by key to produce counts
+        JavaPairRDD<String, Integer> counts = wordOnePairs.reduceByKey(
+                new Function2<Integer, Integer, Integer>() {
+                    public Integer call(Integer x, Integer y) {
+                        return x + y;
+                    }
+                }
+        );
+```
+
+**Output with RDD action saveAsTextFile**
+
+Finally, the RDD action saveAsTextFile(path) writes the elements of the dataset as a text file (or set of text files) in the outputFile directory.
+
+```Java
+        String outputFile = args[1];
+        // Save the word count back out to a text file, causing evaluation.
+        counts.saveAsTextFile(outputFile);                
+```
+
+**Example Word Count App in Scala**
+
+Here is the same example in Scala:
+
+![](/img/tutorial_spark8.png)
```

Comment:
This image looks rather fuzzy, too. You might also consider cropping the left and top a little to get rid of those lines:

![image](https://user-images.githubusercontent.com/77496261/137520563-71a00c22-a4d1-4423-84e5-aa6e996805fe.png)